File: cpp/lv1/budget.cpp
// 예산
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

/* Answer 1.
int solution(vector<int> d, int budget) {
    int sum = 0;
    int count = 0;
    
    sort(d.begin(), d.end());
    
    for(int i = 0; i < d.size(); i++)
    {
        sum += d[i];
        if(sum <= budget)
        {
            count++;
        }
        else
            break;
    }
    return count;
} */

/* Answer 2.
int solution(vector<int> d, int budget) {
    int answer = 0;
    sort(d.begin(), d.end());
    for(int i = 0; i < d.size(); i++)
        if(budget - d[i] >= 0){
            budget -= d[i];
            answer++;
        }
    return answer;
} */

/* Answer 3.
int solution(vector<int> d, int budget) {
    int answer = 0;
    sort(d.begin(), d.end());

    for (int i = 0; i < d.size(); i++)
    {
        if (d[i] > budget)
            break;
        budget -= d[i];
        answer++;
    }
    return answer;
} */

File: python/lv1/budget.py
# 예산
'''
Answer 1.
def solution(d, budget):
    answer = 0
    d.sort()
    for item in d:
        if item > budget:
            break
        else:
            budget -= item
            answer += 1
    
    return answer
'''

'''
Answer 2.
def solution(d, budget):
    d.sort()
    while budget < sum(d):
        d.pop()
    return len(d)

'''
